civis/io/_files.py

```diff
@@ -242,6 +228,21 @@ def file_to_civis(buf, name, api_key=None, client=None, **kwargs):
         with open(buf, 'rb') as f:
             return _file_to_civis(
                 f, name, api_key=api_key, client=client, **kwargs)
+
+    # if buf is not a file handle or if current position is not 0
+    # then write to a file
+    if not isinstance(buf, io.BufferedReader) or buf.tell() != 0:
+        with TemporaryDirectory() as tmp_dir:
+            tmp_path = os.path.join(tmp_dir, 'file_to_civis.csv')
+            try:
```

Comment:
Let me know if you have any ideas on a more elegant way of doing this. I found that buffers produced by some modules, such as zipfile, have a mode attribute that is misleading in that the mode is 'r' but only bytes are ever returned. I figured this was the safest way given we don't know what mode the buffer was opened in.